UnityProject/Assets/Scripts/SpriteHandler/SpriteHandler.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Networking;
+
+public class SpriteHandler : NetworkBehaviour
+{
+	public SpriteRenderer spriteRenderer;
+
+	public List<SpriteList> spriteList = new List<SpriteList>();
+	[SyncVar(hook = nameof(SyncSprite))] public int spriteIndex;
+
+	private List<SpriteInfo> infoList = new List<SpriteInfo>();
+	private int animationIndex = 0;
+	private float timeElapsed = 0;
+	private float waitTime;
+
+	public override void OnStartClient()
+	{
+		SyncSprite(spriteIndex);
+	}
+
+	public void UpdateMe()
+	{
+		timeElapsed += Time.deltaTime;
+		if (timeElapsed >= waitTime)
+		{
+			animationIndex++;
+			if (animationIndex >= infoList.Count)
+			{
+				animationIndex = 0;
+			}
+			SetSprite(infoList[animationIndex]);
+		}
+	}
+
+	void SetSprite(SpriteInfo animationStills)
+	{
+		timeElapsed = 0;
+		waitTime = animationStills.waitTime;
+		spriteRenderer.sprite = animationStills.sprite;
+	}
+
+	void SetSpriteList(List<Sprite> newSprites)
+	{
+		infoList = new List<SpriteInfo>();
+		for (int i = 0; i < newSprites.Count; i++)
+		{
+			var newSprite = new SpriteInfo(newSprites[i], 0.1f);
+			infoList.Add(newSprite);
+		}
+		if (newSprites.Count > 1)
+		{
+			UpdateManager.Instance.Add(UpdateMe);
+		}
+		else
+		{
+			UpdateManager.Instance.Remove(UpdateMe);
+		}
+		SetSprite(infoList[0]);
+	}
+
+	public void ChangeSprite(int newSprites)
+	{
+		if (spriteIndex != newSprites)
+		{
+			spriteIndex = newSprites;
+		}
+	}
+
+	void SyncSprite(int value)
+	{
+		SetSpriteList(spriteList[value].spriteList);
+	}
+
+	[System.Serializable]
```

Comment:
Why is this needed? I think it can be removed. This makes me think you are trying to send this class over the network, which is not possible (Sprite cannot be serialized, pretty sure)

Reply:
it's so the inspector can view The class properly , If it's not there sometimes it just doesn't show up at all